lib/util/http/index.ts

```diff
@@ -162,24 +163,32 @@ export class Http<Opts extends HttpOptions = HttpOptions> {
     // istanbul ignore else: no cache tests
     if (!resPromise) {
       const startTime = Date.now();
-      const queueTask = (): Promise<Response<T>> => {
+      const httpTask = (): Promise<Response<T>> => {
         const queueDuration = Date.now() - startTime;
-        return gotRoutine(url, options, {
+        return gotTask(url, options, {
           method: options.method ?? 'get',
           url,
           queueDuration,
         });
       };
+
+      const throttle = getThrottle(url);
+      const throttledTask = (): Promise<Response<T>> =>
+        throttle?.add<Response<T>>(httpTask) ?? httpTask();
+
       const queue = getQueue(url);
-      resPromise = queue?.add(queueTask) ?? queueTask();
+      const queuedTask = (): Promise<Response<T>> =>
+        queue?.add<Response<T>>(throttledTask) ?? throttledTask();
+
+      resPromise = queuedTask();
       if (options.method === 'get' || options.method === 'head') {
         memCache.set(cacheKey, resPromise); // always set if it's a get or a head
       }
     }
 
     try {
       const res = await resPromise;
-      res.authorization = !!options?.headers?.authorization;
+      (res as any).authorization = !!options?.headers?.authorization;
```

Comment:
we should cast res instead

Reply:
To which type?

Reply:
should probably `HttpResponse<T>` ? it's the function return type.

Reply:
Hah, good catch